### DISCUSSION TITLE
The GLB model with the picture is not displayed

### DISCUSSION THREAD

**Discussion by ttyomaa6** [google/filament#6006] [Q&A]
I load regular models of different colors from the server into the scene and they display well, but when I try to load a model that has a texture with a picture, it just simply does not display in the scene and does not show any errors and exceptions, I have tried to display this GLB model in the different viewers and there I had no problems and everything was displayed, I do not know what could be wrong, I have seen examples of adding textures to the model in filament, but there were separate file, and I have a file like GLB, where, as I understand, the textures are stored when exporting

**Comment by ttyomaa6** [comment]
At the moment I miss that the export model from Blender is not correct and textures and images are not saved, does anyone have experience in this matter?

**Comment by ttyomaa6** [comment]
I found someone else's model with a picture on the Internet and it is displayed in the scene as it should, it turns out it is in the my export model, I understand that the question is not quite appropriate, but I still think that experts here once had to deal with the blender and maybe still someone has faced this problem

**Comment by bejado** [comment]
Are you able to upload your .blend file / glTF file?

**Reply by ttyomaa6** [reply]
GLB file link in goolge disk:
https://drive.google.com/file/d/1vWPgA4PdhieZRC7jB0iqe2jAro562t3h/view?usp=drivesdk

.blend file link in google disk:
https://drive.google.com/file/d/1vRr10vhvLMx8-KpBKjxbrad6ZO9ll57O/view?usp=drivesdk

**Reply by bejado** [reply]
I've requested access.

**Reply by ttyomaa6** [reply]
access is open, i hope you can help me

**Reply by bejado** [reply]
Here's what I see when I view your glb file:

<img width="568" alt="Screen Shot 2022-08-31 at 3 00 06 PM" src="https://user-images.githubusercontent.com/5298046/187761182-f90573e8-6dc5-4af3-b7be-2784800dbaf0.png">

Is that what you expect?

Some notes:
- Your .blend file doesn't have the textures packed, so I can't verify if the glb matches Blender.
- Your materials in Blender have additional nodes, like "Hue Saturation Value" and "RGB Curves". I'm not sure if Blender's glTF exporter will handle those correctly.

**Reply by ttyomaa6** [reply]
when I tried to load this model into the scene, my application crashed

here is a file and a picture of another model, when I try to view it in the scene, my application also crashes

![IMG_20220831_222512_733.jpg](https://user-images.githubusercontent.com/81015461/187765883-70326cbe-ac27-44c6-be96-5d6d0ea97b59.jpg)

gave you access

https://drive.google.com/file/d/1vlEo24MCHTb6n3nf8RqTZW9WbR0dQY6g/view?usp=drivesdk

and can you tell me how to export models from blender correctly if you know

**Reply by bejado** [reply]
What logs are printed when your application crashes?

**Reply by ttyomaa6** [reply]
I myself would like to know the root of the problem, but in the logs I only get terminating

I want to ask a couple more questions.  I have already read a lot of articles and saw that when creating materials, everyone uses glTF Material Roughness in the material tree, and I use Principled BSDF, I also apply PNG images as texters, although many use pbr format materials, do I have errors on  this stage?

**Reply by ttyomaa6** [reply]
i tried to download the glb sample model
 https://github.com/KhronosGroup/glTF-Sample-Models/blob/master/2.0/WaterBottle/glTF-Binary/WaterBottle.glb

 and sent it to blender, then exported it from there and sent it  into the application where I loaded the scene, but I didn’t see the object, I decided to try to leave only the base color without other textures and it worked, could something be wrong with my blender?

I could check this problem if you sent me any model with texture that you exported from blender, although maybe I'm already asking a lot

**Reply by ttyomaa6** [reply]
Now my partners and I are doing it like this

![addons_io-gltf2_material-baseColor-imageHookup.png](https://user-images.githubusercontent.com/81015461/187795176-1bbb086f-c8f7-4f2a-b633-2da7f5a281e5.png)


do we need to do it this way and what is the difference?

![addons_io-gltf2_material-mapping.png](https://user-images.githubusercontent.com/81015461/187795191-891a2f96-a263-4739-87f9-1dcd7fb99185.png)

**Reply by ttyomaa6** [reply]
?

**Reply by bejado** [reply]
I don't know the specifics of Blender's glTF exporter addon, you'll have to try asking at https://github.com/KhronosGroup/glTF-Blender-IO.

I did not have any issues loading the earlier model you uploaded, so there could be something wrong in your code. You can try viewing your model in our [gltf_viewer app](https://github.com/google/filament/blob/main/samples/gltf_viewer.cpp).
